internal/provider/resource_gitlab_runner_test.go

```diff
@@ -0,0 +1,150 @@
+			{
+				ResourceName:      "gitlab_runner.this",
+				ImportState:       true,
+				ImportStateVerify: true,
+				//These need to be ignored since they don't come back in the "get" command
+				ImportStateVerifyIgnore: []string{"authentication_token", "registration_token"},
+			},
+			{
+				Config: fmt.Sprintf(`
+				resource "gitlab_runner" "this" {
+					registration_token = "%s"
+					description = "Lorem Ipsum Dolor Sit Amet"
+				}
+				`, group.RunnersToken),
+			},
+			{
+				ResourceName:      "gitlab_runner.this",
+				ImportState:       true,
+				ImportStateVerify: true,
+				//These need to be ignored since they don't come back in the "get" command
+				ImportStateVerifyIgnore: []string{"authentication_token", "registration_token"},
+			},
+		},
+	})
+}
+
+func TestAccGitlabRunner_comprehensive(t *testing.T) {
+	testAccCheck(t)
+
+	group := testAccCreateGroups(t, 1)[0]
+	//The runner token is not populated on the return from the group create, so re-retrieve it to get the token.
+	group, _, err := testGitlabClient.Groups.GetGroup(group.ID, &gitlab.GetGroupOptions{})
+	if err != nil {
+		t.Fail()
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() {},
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckRunnerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`
+				resource "gitlab_runner" "this" {
+					registration_token = "%s"
+					description = "Lorem Ipsum"
+
+					paused = false
+					locked = false
+					run_untagged = false
+					tag_list = ["tag_one", "tag_two"]
+					access_level = "ref_protected"
+					maximum_timeout = 3600					
+				}
+				`, group.RunnersToken),
+			},
+			{
+				ResourceName:      "gitlab_runner.this",
+				ImportState:       true,
+				ImportStateVerify: true,
+				//These need to be ignored since they don't come back in the "get" command
+				ImportStateVerifyIgnore: []string{"authentication_token", "registration_token"},
+			},
+			{
+				Config: fmt.Sprintf(`
+				resource "gitlab_runner" "this" {
+					registration_token = "%s"
+					description = "Lorem Ipsum Dolor Sit Amet"
+
+					paused = true
+					locked = true
+					run_untagged = true
+					tag_list = ["tag_one", "tag_two", "tag_three"]
+					access_level = "not_protected"
+					maximum_timeout = 4200
+				}
+				`, group.RunnersToken),
+			},
+			{
+				ResourceName:      "gitlab_runner.this",
+				ImportState:       true,
+				ImportStateVerify: true,
+				//These need to be ignored since they don't come back in the "get" command
+				ImportStateVerifyIgnore: []string{"authentication_token", "registration_token"},
+			},
+		},
+	})
+}
+
+func testAccCheckRunnerDestroy(state *terraform.State) error {
+	for _, rs := range state.RootModule().Resources {
+		if rs.Type != "gitlab_runner" {
+			continue
+		}
+
+		id, _ := strconv.Atoi(rs.Primary.ID)
+
+		runner, _, err := testGitlabClient.Runners.GetRunnerDetails(id)
+		if err == nil {
+			if runner != nil && runner.ID == id {
```

Comment:
I know a lot of these destroy check functions have that check and I don't understand why, thus, I usually remove / don't copy them:

```suggestion
			if runner != nil {
```

Reply:
Works for me 😄  - I'm going to squash commits again here in a bit, but going to try and see if I can get a helm chart example and doing some research on the instance runner token set first.